File: jsimple-oauth/src/main/java/jsimple/oauth/builder/api/GoogleOauthApi.java
package jsimple.oauth.builder.api;

import jsimple.oauth.model.Token;
import jsimple.oauth.model.Verb;

public class GoogleOAuthApi extends DefaultOAuthApi10a {
    @Override public String getAccessTokenEndpoint() {
        return "https://www.google.com/accounts/OAuthGetAccessToken";
    }

    @Override public String getRequestTokenEndpoint() {
        return "https://www.google.com/accounts/OAuthGetRequestToken";
    }

    @Override public String getAccessTokenVerb() {
        return Verb.GET;
    }

    @Override public String getRequestTokenVerb() {
        return Verb.GET;
    }

    @Override public String getAuthorizationUrl(Token requestToken) {
        return "https://www.google.com/accounts/OAuthAuthorizeToken?oauth_token=" + requestToken.getTokenString();
    }
}

File: jsimple-oauth/c#/jsimple/oauth/builder/api/GoogleOAuthApi.cs
namespace jsimple.oauth.builder.api
{

	using Token = jsimple.oauth.model.Token;
	using Verb = jsimple.oauth.model.Verb;

	public class GoogleOAuthApi : DefaultOAuthApi10a
	{
		public override string AccessTokenEndpoint
		{
			get
			{
				return "https://www.google.com/accounts/OAuthGetAccessToken";
			}
		}

		public override string RequestTokenEndpoint
		{
			get
			{
				return "https://www.google.com/accounts/OAuthGetRequestToken";
			}
		}

		public override string AccessTokenVerb
		{
			get
			{
				return Verb.GET;
			}
		}

		public override string RequestTokenVerb
		{
			get
			{
				return Verb.GET;
			}
		}

		public override string getAuthorizationUrl(Token requestToken)
		{
			return "https://www.google.com/accounts/OAuthAuthorizeToken?oauth_token=" + requestToken.TokenString;
		}
	}

}